presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -605,27 +604,22 @@ public static String computeBucketedFileName(String filePrefix, int bucket)
         return filePrefix + "_bucket-" + Strings.padStart(Integer.toString(bucket), BUCKET_NUMBER_PADDING, '0');
     }
 
-    public static String getFileExtension(JobConf conf, StorageFormat storageFormat)
+    public static String getFileExtension(StorageFormat storageFormat, HiveCompressionCodec compressionCodec)
     {
         // text format files must have the correct extension when compressed
-        if (!HiveConf.getBoolVar(conf, COMPRESSRESULT) || !HiveIgnoreKeyTextOutputFormat.class.getName().equals(storageFormat.getOutputFormat())) {
+        if (compressionCodec == NONE || !HiveIgnoreKeyTextOutputFormat.class.getName().equals(storageFormat.getOutputFormat())) {
             return "";
         }
 
-        String compressionCodecClass = conf.get("mapred.output.compression.codec");
-        if (compressionCodecClass == null) {
+        if (!compressionCodec.getCodec().isPresent()) {
             return new DefaultCodec().getDefaultExtension();
         }
 
         try {
-            Class<? extends CompressionCodec> codecClass = conf.getClassByName(compressionCodecClass).asSubclass(CompressionCodec.class);
-            return ReflectionUtil.newInstance(codecClass, conf).getDefaultExtension();
+            return compressionCodec.getCodec().get().getConstructor().newInstance().getDefaultExtension();
         }
-        catch (ClassNotFoundException e) {
-            throw new PrestoException(HIVE_UNSUPPORTED_FORMAT, "Compression codec not found: " + compressionCodecClass, e);
-        }
-        catch (RuntimeException e) {
-            throw new PrestoException(HIVE_UNSUPPORTED_FORMAT, "Failed to load compression codec: " + compressionCodecClass, e);
+        catch (ReflectiveOperationException e) {
```

Comment:
Does that mean `RuntimeException`  caught in the original code is too broad?  (i.e. it should always catch `ReflectiveOperationException`)

Reply:
I think so. I manually checked the constructors, and those constructors don't throw any exceptions.